diff --git a/pyproject.toml b/pyproject.toml
index b4dedf4..160e982 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -22,8 +22,9 @@ test-command = "pytest --exitfirst --verbose --failed-first {package}/tests"
 before-all = [
     # Has eigen3-devel.aarch64
     "yum install -y epel-release",
-    # "yum install -y wget freetype-devel libpng12-devel pixman-devel zlib-devel eigen3-devel",
-    "yum install -y wget freetype-devel libpng12-devel pixman-devel zlib-devel",
+    "yum install -y wget freetype-devel libpng12-devel pixman-devel zlib-devel eigen3-devel",
+    "yum install -y lapack-devel openblas-devel",
+    # "yum install -y wget freetype-devel libpng12-devel pixman-devel zlib-devel",
     # cairo-devel from the centos repo does not work, build here
     "wget https://www.cairographics.org/snapshots/cairo-1.15.14.tar.xz --no-check-certificate",
     "tar xvf cairo-*",
diff --git a/setup.py b/setup.py
index e915938..d8e05e8 100644
--- a/setup.py
+++ b/setup.py
@@ -168,6 +168,18 @@ class BuildRDKit(build_ext_orig):
             ["git", "clone", "-b", f"{ext.rdkit_tag}", "https://github.com/rdkit/rdkit"]
         )
 
+        # patch and copy files to add Osmordred (also backport font sha fix)
+        osmordred_patches_path = Path.resolve(Path.joinpath(cwd, "osmordred_rdkit_2023_09_3_patches"))
+        osmordred_source_path = Path.resolve(Path.joinpath(cwd, "osmordred_source"))
+        check_call(
+            f"echo PATCHING Osmordred from {osmordred_patches_path}; cd rdkit; git apply {osmordred_patches_path}/*.diff",
+            shell=True,
+        )
+        check_call(
+            f"echo COPYING Osmordred sources {osmordred_source_path}; cp -a {osmordred_source_path}/* rdkit/",
+            shell=True,
+        )
+
         # Location of license file
         license_file = build_path / "rdkit" / "license.txt"
 
@@ -201,7 +213,8 @@ class BuildRDKit(build_ext_orig):
             "-DRDK_BUILD_CPP_TESTS=OFF",
             # Fix InChi download
             "-DINCHI_URL=https://rdkit.org/downloads/INCHI-1-SRC.zip",
-            
+            # Build Osmordred code
+            "-DRDK_BUILD_OSMORDRED=ON",
         ]
 
         # Modifications for Windows
@@ -343,7 +356,7 @@ class BuildRDKit(build_ext_orig):
 
 setup(
     name="rdkit",
-    version=rdkit_tag.replace("Release_", "").replace("_", "."),
+    version=rdkit_tag.replace("Release_", "").replace("_", ".") + "+osmordred",
     description="A collection of chemoinformatics and machine-learning software written in C++ and Python",
     author="Christopher Kuenneth",
     author_email="chris@kuenneth.dev",
