diff --git a/CMakeLists.txt b/CMakeLists.txt
index cb79b16e8..5c9f44083 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,6 +70,7 @@ option(RDK_BUILD_FUZZ_TARGETS "build the fuzz targets" OFF)
 option(RDK_BUILD_MINIMAL_LIB_RXN "build support for reactions into MinimalLib" ON )
 option(RDK_BUILD_MINIMAL_LIB_SUBSTRUCTLIBRARY "build support for SubstructLibrary into MinimalLib" ON )
 option(RDK_BUILD_MINIMAL_LIB_MCS "build support for MCS into MinimalLib" OFF )
+option(RDK_BUILD_OSMORDRED "build Osmordred functions" OFF)
 
 set(RDK_BOOST_VERSION "1.58.0")
 
@@ -625,6 +626,11 @@ if(RDK_BUILD_CONTRIB)
   add_subdirectory(Contrib)
 endif(RDK_BUILD_CONTRIB)
 
+if(RDK_BUILD_OSMORDRED)
+  set(BLA_VENDOR OpenBLAS)
+  find_package(LAPACK REQUIRED)
+endif(RDK_BUILD_OSMORDRED)
+
 # export the project targets (to be included in the cmake package configuration)
 install(
   EXPORT ${RDKit_EXPORTED_TARGETS}
