diff --git a/Code/GraphMol/MolDraw2D/CMakeLists.txt b/Code/GraphMol/MolDraw2D/CMakeLists.txt
index 6cd0833a2..58a8a3ebc 100644
--- a/Code/GraphMol/MolDraw2D/CMakeLists.txt
+++ b/Code/GraphMol/MolDraw2D/CMakeLists.txt
@@ -17,14 +17,16 @@ if (RDK_BUILD_FREETYPE_SUPPORT AND RDK_INSTALL_COMIC_FONTS)
         set(needDownload "FALSE")
     endif ()
     if (needDownload)
-        set(MD5Sum "850b0df852f1cda4970887b540f8f333")
-        downloadAndCheckMD5("https://fonts.google.com/download?family=Comic%20Neue"
-                "${CMAKE_CURRENT_SOURCE_DIR}/Comic_Neue.zip"
-                ${MD5Sum})
-        execute_process(COMMAND ${CMAKE_COMMAND} -E tar x
-                ${CMAKE_CURRENT_SOURCE_DIR}/Comic_Neue.zip --format=zip
-                WORKING_DIRECTORY ${RDKit_DataDir}/Fonts)
+        # we started having problems with constantly changing MD5s on the zip file,
+        # so now we just check the MD5 of the target file
+        downloadAndCheckMD5("https://github.com/google/fonts/raw/main/ofl/comicneue/ComicNeue-Regular.ttf"
+                "${RDKit_DataDir}/Fonts/ComicNeue-Regular.ttf"
+                "fc1eac54b325542d4c133732658f823b")
+        downloadAndCheckMD5("https://github.com/google/fonts/raw/main/ofl/comicneue/OFL.txt"
+                "${RDKit_DataDir}/Fonts/OFL.txt"
+                "")
     endif (needDownload)
+
 endif ()

 rdkit_headers(MolDraw2D.h
